pool.go

```diff
@@ -195,9 +210,23 @@ func (p *AuthProviderPool) Provision(name string) error {
 		m.TokenIssuer = master.TokenIssuer
 	}
 
-	if m.TokenSecret == "" {
-		m.TokenSecret = master.TokenSecret
+	if m.TokenHSSecret == "" {
+		m.TokenHSSecret = master.TokenHSSecret
+	}
+	if err := loadRSKeys(m); err != nil {
```

Comment:
@njones , noticed you passing pointers to functions. I think in this case you should consider creating a method `m.LoadEncryptionKeys()`.

Reply:
Sure, no problem... do you think `LoadEncryptionKeys` should load both `RSA` and `EC` keys, assuming that `EC` encryption gets added, or is it too much of a YAGNI ... and we just change it when EC gets added --if ever?